src/main/scala/com.mediative.eigenflow/process/ProcessSupervisor.scala

```diff
@@ -0,0 +1,47 @@
+package com.mediative.eigenflow.process
+
+import java.util.Date
+
+import akka.actor.SupervisorStrategy.Stop
+import akka.actor.{ Actor, ActorRef, OneForOneStrategy, Props, SupervisorStrategy, Terminated }
+import com.mediative.eigenflow.StagedProcess
+import com.mediative.eigenflow.publisher.MessagingSystem
+
+import scala.util.{ Failure, Success, Try }
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object ProcessSupervisor {
+  def props(process: StagedProcess, date: Option[Date])(implicit messagingSystem: MessagingSystem) = Props(new ProcessSupervisor(process, date))
+
+  case object Done
+  case object Failed
+}
+
+class ProcessSupervisor(val process: StagedProcess, val startDate: Option[Date])(implicit val messagingSystem: MessagingSystem) extends Actor {
```

Comment:
This might work.  Not sure, however it doesn't feel right to me.  Relying on a System.exit feels like a very brute force solution.

I suspect that we need to rethink the structure of the actors and how they handle failure.  I think maybe at the top level, where we create the Actor System as well as any other connections, we need a shutdown hook.  This hook simply needs to be called in the event of a completion or failure.

We might need a single top level actor that has access to that shutdown hook so it can call it.  That top level actor is responsible for creating all other actors.  So in the event of a failure we can always escalate back to that single top level actor who can then shutdown as appropriate.

Not sure exactly what this looks like, but I don't really like having to do a System.exit() as it doesn't give us the opportunity to run any necessary cleanup code.

I think this supervisor actor could be reworked to be that top level actor.  The key is that instead of calling System.exit, it should call some "shutdown" function.  That function can perform important steps like stopping the messaging system, but also terminating the actor system etc.  However, that function should be defined in the same place that the messaging system and the actor system are created (ideally).